x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/InferenceParams.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class InferenceParams implements ToXContentObject, Writeable {
+
+    public static ParseField NUM_TOP_CLASSES = new ParseField("num_top_classes");
+
+    public static InferenceParams EMPTY_PARAMS = new InferenceParams(0);
+
+    private final int numTopClasses;
```

Comment:
Now we are coupling `InferenceParams` to classification since those are part of the request object we'll have BWC issues if we change this in the future. I think we should consider whether we should have a `InferenceConfig` named writeable which lets us do this more nicely plus it gives us a chance to sanity check the model id matches the inference type the user expects.

Reply:
I agree, I think this may clean up some execution paths here and there. Additionally, it will allow us to do the sanity check you mention. 

I will complete this in a follow up PR. It would definitely add more LOC churn and this PR is already beefy.

Reply:
@dimitris-athanasiou I wrote the changes up and am ready to open a new PR as soon as this one is closed for this change. It would have added ~200 LOC to this PR.

Reply:
Great!